tests/test_samples.py

```diff
@@ -0,0 +1,65 @@
+"""Snapshot test of formatting of sample files.
+
+Code samples are located in tests/samples. To add a new sample create a new
+file and create a snapshot of the formatting result with
+
+    $ pytest --snapshot-update
+
+Commit both the sample as well as the updated snapshot.
+"""
+
+import io
+from pathlib import Path
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+from syrupy.extensions.single_file import SingleFileSnapshotExtension
+
+from testutils import zeekscript
+
+SAMPLES_DIR = Path(__file__).parent / "samples"
+
+
+# Use a custom snapshot fixture so we emit one file per generated test case
+# instead of one per module.
+@pytest.fixture
+def snapshot(snapshot: SnapshotAssertion):  # pylint: disable=redefined-outer-name
+    return snapshot.use_extension(SingleFileSnapshotExtension)
+
+
+def _format(script: zeekscript.Script):
+    """Formats a given `Script`"""
+    buf = io.BytesIO()
+    script.format(buf)
+    return buf.getvalue().decode()
```

Comment:
I see you put `encode` `utf8` below. I had dropped the `.decode()` here when testing :-) Slight preference for not doing an extra decode/encode roundtrip.

Reply:
Cleaned up, good idea.